portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,36 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays; 
+import com.google.gson.Gson;
+import java.util.*;
+import com.google.common.util.concurrent.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    final String urlRedirect = "https://8080-dot-12380874-dot-devshell.appspot.com/";
+    List<String> comments = new ArrayList<>();
+    RateLimiter rateLimiter = RateLimiter.create(1);
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello from Chris!!");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    	response.setContentType("application/json;");
+        Gson gson = new Gson();
+        response.getWriter().println(gson.toJson(comments));
+    }
+
+	@Override
+  	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json;");
+        boolean isAcquired = rateLimiter.tryAcquire();
+        if(!isAcquired) {
```

Comment:
Please explain with a comment why you need a rate limiter and what will happen if it is not acquired

Reply:
Kiran, with PR 3894a90, I added Java comments to the doGet and doPost. 

The Rate Limiter will pause that request if it goes over the limit until the rate limiter resets.

Reply:
Per this document: https://cloud.google.com/datastore/docs/best-practices

and https://cloud.google.com/datastore/docs/concepts/limits

It looks like removing the rate limiter will be the best option.